Lib/enum.py

```diff
@@ -709,18 +716,18 @@ def __call__(cls, value, names=None, *values, module=None, qualname=None, type=N
         """
         if cls._member_map_:
             # simple value lookup if members exist
-            if names:
+            if names is not _not_given:
                 value = (value, names) + values
             return cls.__new__(cls, value)
         # otherwise, functional API: we're creating a new Enum type
-        if names is None and type is None:
+        if names is _not_given and type is None:
             # no body? no data-type? possibly wrong usage
             raise TypeError(
                     f"{cls} has no members; specify `names=()` if you meant to create a new, empty, enum"
                     )
         return cls._create_(
                 class_name=value,
-                names=names,
+                names=names or None,
```

Comment:
Add tests for something like:
* `Enum('empty_enum', '')`
* `Enum('empty_enum', [])`
* `Enum('empty_enum', {})`
* `Enum('empty_enum', '', type=int)`
* `Enum('empty_enum', [], type=int)`
* `Enum('empty_enum', {}, type=int)`

Reply:
Done.